airbyte_cdk/sources/declarative/models/declarative_component_schema.py

```diff
@@ -1831,6 +1831,21 @@ class DynamicSchemaLoader(BaseModel):
         description="Component used to coordinate how records are extracted across stream slices and request pages.",
         title="Retriever",
     )
+    schema_transformations: Optional[
+        List[
+            Union[
+                AddFields,
+                CustomTransformation,
+                RemoveFields,
+                KeysToLower,
+                KeysToSnakeCase,
```

Comment:
> Can this list be moved to another definition? We would need to maintain fewer locations as DeclarativeStream and DynamicSchemaLoader now have transformations.

Actually, this is the right place for the comment, right? As yaml is automatically generated.

Reply:
For example, we guess we want the "flatten fields" transformation to be added here, so we just do it once on a common list.

Reply:
You are correct; this is an autogenerated file. Therefore, all efforts to unify the interface should be handled in the YAML schema file, which could lead to more complex logic or misleading as not all components from list of refs can be support by parent component . As such, it's best to leave this as it is and bring up this issue during our next brainstorming session.